tftb/processing/linear.py

```diff
@@ -163,11 +165,28 @@ def gabor(signal, n_coeff=None, q_oversample=None, window=None):
     return tfr, dgr, gam
 
 
-if __name__ == '__main__':
+def main():
     from tftb.generators import fmconst
     import matplotlib.pyplot as plt
+
     sig = np.r_[fmconst(128, 0.2)[0], fmconst(128, 0.4)[0]]
-    ts = np.linspace(0, 1, 256)
-    tfr = ShortTimeFourierTransform(sig, timestamps=ts)
-    tfr.run()
-    tfr.plot(show_tf=True, cmap=plt.cm.viridis)
+
+    nr_samples = 256
+    n_fbins = nr_samples
+    nperseg = 65
+    noverlap = nperseg - 1
+    window = hamming(nperseg)
+    nfft = 256
```

Comment:
Are all of these variables really needed to be declared here? The `ShortTimeFourierTransform` class should automatically compute them by default. These options should be passed only if we need to _override_ these defaults.

For example, the `n_fbins` param will always be set to the length of the signal by default, internally by the STFT class. Why add it here again? Similarly the default window is always Hamming, with `nperseg`, which in turn always defaults to `n_fbins / 4` (+1 if it's an even number).

Reply:
See my comment further up.
The defaults you are providing are different to those provided by Scipy.
I would prefer not to define parallel defaults as this could be confusing.

If you really do want to provide new defaults, then please let me know what defaults you would like.